kratos/mpi/tests/sources/test_pointer_communicator.cpp

```diff
@@ -186,9 +187,81 @@ KRATOS_TEST_CASE_IN_SUITE(PointerCommunicatorConstructByFunctor, KratosMPICoreFa
     }
 
 
-};
+}
+
+KRATOS_TEST_CASE_IN_SUITE(PointerMapCommunicatorAssembly, KratosMPICoreFastSuite)
+{
+    DataCommunicator& r_default_comm = ParallelEnvironment::GetDefaultDataCommunicator();
+    Model current_model;
+    auto& mp = current_model.CreateModelPart("mp");
+    mp.AddNodalSolutionStepVariable(PARTITION_INDEX);
+
+    const int world_size = r_default_comm.Size();
+    const int current_rank = r_default_comm.Rank();
+
+    auto pnode = mp.CreateNewNode(current_rank+1, current_rank,current_rank,current_rank); //the node is equal to the current rank;
+    pnode->FastGetSolutionStepValue(PARTITION_INDEX) = current_rank;
+    pnode->SetValue(TEMPERATURE, current_rank);
+    pnode->SetValue(VELOCITY, array_1d<double, 3>(3, current_rank));
+
+    // creates the gp map
+    GlobalPointerMapCommunicator<Node<3>, double> pointer_map_double_comm(r_default_comm);
+    GlobalPointerMapCommunicator<Node<3>, array_1d<double, 3>> pointer_map_array3_comm(r_default_comm);
 
+    // creates apply proxy
+    auto apply_temperature_assemble_proxy = pointer_map_double_comm.GetApplyProxy(
+        [](Node<3>& rNode, const double& NewValue) {
```

Comment:
it should also be possible to fuse this two operations into one, this way you have just one communication. i mean something on the line of 

~~~
class MyCustomData
{
     public:
     double temperature
     array_1d<double,3> velocity;

     private:
     save(...)
     load(...)
}
 GlobalPointerMapCommunicator<Node<3>, myCustomData>
~~~

so that the full MyCustomData is sent via serialization (the communicator should detect automatically the optimized code path for simple mpi data types)

Reply:
This is totally allowed. I had this `VELOCITY` assembly merely to check for something other than primitive data types which has the serialization methods defined. :)